src/istio/control/http/attributes_builder.cc

```diff
@@ -128,12 +128,18 @@ void AttributesBuilder::ExtractAuthAttributes(CheckData *check_data) {
     builder.AddStringMap(utils::AttributeName::kRequestAuthClaims, payload);
   }
   std::string source_user;
-  if (check_data->GetSourceUser(&source_user)) {
+  if (check_data->GetPrincipal(true, &source_user)) {
     // TODO(diemtvu): remove kSourceUser once migration to source.principal is
     // over. https://github.com/istio/istio/issues/4689
     builder.AddString(utils::AttributeName::kSourceUser, source_user);
     builder.AddString(utils::AttributeName::kSourcePrincipal, source_user);
   }
+
+  std::string destination_principal;
```

Comment:
We have an early return at line 107, so this won't be reached if authn filter output available (which should be the case when mTLS or JWT was used). Is that an expected behavior?

FYI, the fallback block (everything from line 110 to 138) should be removed, as we expected authn filter is always added to filter chain when authn is in used.

Reply:
How does JWT carry destination.principal? Is JWT also used for mutual authentication?